test/samples/if_assigned_primitive.instrumented.sol

```diff
@@ -55,91 +55,117 @@ contract Base {
     }
 
     function Base_x_inline_initializer() internal {
-        if (!(x >= 1)) {
-            emit AssertionFailed("0: ");
-            assert(false);
+        unchecked {
+            if (!(x >= 1)) {
+                emit AssertionFailed("0: ");
+                assert(false);
+            }
         }
     }
 
     function Base_y_inline_initializer() internal {
         vars1 memory _v;
-        _v.old_0 = y;
-        if (!(y >= _v.old_0)) {
-            emit AssertionFailed("1: ");
-            assert(false);
+        unchecked {
+            _v.old_0 = y;
+        }
+        unchecked {
+            if (!(y >= _v.old_0)) {
+                emit AssertionFailed("1: ");
+                assert(false);
+            }
         }
     }
 
     function Base_a_inline_initializer() internal {
-        if (!(uint160(a) >= 1)) {
-            emit AssertionFailed("2: ");
-            assert(false);
+        unchecked {
+            if (!(uint160(a) >= 1)) {
+                emit AssertionFailed("2: ");
+                assert(false);
+            }
         }
     }
 
     function Base_x_uint256_assign(uint256 ARG0) internal returns (uint256 RET0) {
         x = ARG0;
         RET0 = x;
-        if (!(x >= 1)) {
-            emit AssertionFailed("0: ");
-            assert(false);
+        unchecked {
+            if (!(x >= 1)) {
+                emit AssertionFailed("0: ");
+                assert(false);
+            }
         }
     }
 
     function Base_a_address_assign(address ARG1) internal returns (address RET1) {
         a = ARG1;
         RET1 = a;
-        if (!(uint160(a) >= 1)) {
-            emit AssertionFailed("2: ");
-            assert(false);
+        unchecked {
+            if (!(uint160(a) >= 1)) {
+                emit AssertionFailed("2: ");
+                assert(false);
+            }
         }
     }
 
     function Base_y_uint256_assign(uint256 ARG2) internal returns (uint256 RET2) {
         vars5 memory _v;
-        _v.old_1 = y;
+        unchecked {
+            _v.old_1 = y;
+        }
         y = ARG2;
         RET2 = y;
-        if (!(y >= _v.old_1)) {
-            emit AssertionFailed("1: ");
-            assert(false);
+        unchecked {
+            if (!(y >= _v.old_1)) {
+                emit AssertionFailed("1: ");
+                assert(false);
+            }
         }
     }
 
     function Base_x_inc_postfix() internal returns (uint256 RET3) {
         RET3 = x;
         x++;
```

Comment:
We can have overflow here for `++` and underflow for `--`, I guess. Should we also wrap unary operations (except `delete`) in `unchecked` block?

This is the first occurance, however there are more occurances in other snapshots. I didn't commented them all to not cause comment spamming.

Reply:
So this `++` corresponds to the **original** operation that we are interposing on. It is not wrapped in `unchecked` here because the original operation in `if_assigned_pritimitive.sol` was not wrapped in unchecked https://github.com/ConsenSys/scribble/blob/develop/test/samples/if_assigned_primitive.sol#L25. If you wrap line 25 in if_assigned_primitive.sol in an unchecked, then here `x++` will also be wrapped in unchecked.

Reply:
This however brings a new problem - we may need to emit 2 versions of wrappers - one version when the wrapper is inside of an unchecked, and another when its not inside an unchecked. Good catch!

Reply:
Fixed in 0278d5c